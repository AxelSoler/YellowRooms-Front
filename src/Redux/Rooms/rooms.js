import axios from "axios";

const GET_ROOMS = "details/rooms/GET_ROOMS";

export const getRoomsActionCreator = () => (dispatch) => {
  axios.get("http://localhost:3001/rooms").then((response) => {
    const rooms = response.data;
    dispatch({ type: GET_ROOMS, payload: rooms });
  });
};
const roomsReducer = (state = [], action) => {
  switch (action.type) {
    case GET_ROOMS:
      return action.payload;
    default:
      return state;
  }
};

export default roomsReducer;
