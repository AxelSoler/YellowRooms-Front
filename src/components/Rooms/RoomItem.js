import React from 'react';
import PropTypes from 'prop-types';
import Card from '@mui/material/Card';
import CardContent from '@mui/material/CardContent';
import CardMedia from '@mui/material/CardMedia';
import Typography from '@mui/material/Typography';
import { NavLink } from 'react-router-dom';

const RoomItem = (props) => {
  const { room } = props;

  const imageStyle = {
    height: 0,
    paddingTop: '80%',
    width: '100%',
    objectFit: 'cover',
  };

  const cardStyle = {
    margin: '0 20px',
    height: '100%',
    display: 'flex',
    justifyContent: 'space-between',
    flexDirection: 'column',
    alignItems: 'center',
  };

  return (
    <NavLink
      className="link"
      to={`/Rooms/${room.id}`}
      key={room.id}
    >
      <Card style={cardStyle}>
        <CardMedia image={room.photo} title={room.name} style={imageStyle} />
        <CardContent>
          <Typography gutterBottom variant="h5" component="h2" sx={{ fontWeight: 'bold' }}>
            {room.name}
          </Typography>
          <Typography component="p">{room.description}</Typography>
        </CardContent>
      </Card>
    </NavLink>
  );
};

RoomItem.propTypes = {
  room: PropTypes.shape({
    id: PropTypes.number.isRequired,
    name: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
    photo: PropTypes.string.isRequired,
  }).isRequired,
};

export default RoomItem;
