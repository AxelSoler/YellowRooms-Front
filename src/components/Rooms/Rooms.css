.rooms-container {
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  height: auto;
}

h1 {
  font-size: 3rem;
  font-weight: bold;
}

h4 {
  font-size: 1.5rem;
  margin-bottom: 1rem;
  text-align: center;
}

.carousel-container {
  width: 100%;
  height: 100%;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
}

.carousel-container img {
  width: 100%;
  height: auto;
  object-fit: cover;
}

.carousel-container button {
  width: 1rem;
  padding: 1rem;
  background-color: #92bd04cc;
  color: #fff;
  text-align: center;
}

.carousel-container button:hover {
  background-color: #92bd04;
  transform: scale(1.1);
}
