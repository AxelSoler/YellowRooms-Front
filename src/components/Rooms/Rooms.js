import React from "react";
import { NavLink } from "react-router-dom";
const Rooms = () => {
  const value = `/Rooms/id=${1}`;
  return (
    <>
      <h2>Rooms</h2>
      <NavLink className="link" to={value}>
        Detail
      </NavLink>
    </>
  );
};

export default Rooms;
