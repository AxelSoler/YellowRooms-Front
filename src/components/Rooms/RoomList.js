import React from 'react';
import PropTypes from 'prop-types';
import Carousel from 'react-multi-carousel';
import 'react-multi-carousel/lib/styles.css';
import RoomItem from './RoomItem';

const responsive = {
  desktop: {
    breakpoint: { max: 3000, min: 1024 },
    items: 4,
  },
  laptop: {
    breakpoint: { max: 1024, min: 768 },
    items: 3,
  },
  tablet: {
    breakpoint: { max: 768, min: 464 },
    items: 2,
  },
  mobile: {
    breakpoint: { max: 464, min: 0 },
    items: 1,
  },
};

const RoomList = (props) => {
  const { rooms } = props;

  return (
    <Carousel
      responsive={responsive}
      className="carousel-container"
      infinite
      autoPlay
      autoPlaySpeed={3000}
      keyBoardControl
      customTransition="all .5"
      transitionDuration={500}
      outerSpacing={0}
      autoPlayonDevice="mobile"
      removeArrowOnDeviceType={['tablet', 'mobile']}
    >
      {rooms.map(({ attributes: room }) => (
        <RoomItem key={room.id} room={room} />
      ))}
    </Carousel>
  );
};

RoomList.propTypes = {
  rooms: PropTypes.arrayOf(PropTypes.shape({
    attributes: PropTypes.shape({
      id: PropTypes.number.isRequired,
      name: PropTypes.string.isRequired,
      description: PropTypes.string.isRequired,
      photo: PropTypes.string.isRequired,
    }).isRequired,
  })).isRequired,
};

export default RoomList;
