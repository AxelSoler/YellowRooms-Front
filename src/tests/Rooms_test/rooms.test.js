import React from "react";
import renderer from "react-test-renderer";
import wrapper from "../wrapMeInValidComponents";
import Rooms from "../../components/Rooms/Rooms";

test("Rooms show render the same", () => {
  const jsonRep = renderer.create(wrapper(<Rooms />)).toJSON();
  expect(jsonRep).toMatchSnapshot();
});
