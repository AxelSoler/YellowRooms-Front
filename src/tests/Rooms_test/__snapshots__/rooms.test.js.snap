// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rooms show render the same 1`] = `
Array [
  <section>
    <h1>
      We have the luxuries
    </h1>
    <h4>
      Please select your match
    </h4>
    <div
      className="rooms-container"
    />
  </section>,
  ";",
]
`;
